src/btree/col_srch.c

```diff
@@ -286,11 +298,13 @@ __wt_col_search(
      * because column-store files are dense, but in this case the caller searched past the end of
      * the table.
      */
-    cbt->ins_head = WT_COL_APPEND(page);
-    if ((cbt->ins = __col_insert_search(cbt->ins_head, cbt->ins_stack, cbt->next_stack, recno)) ==
-      NULL)
+    ins_head = WT_COL_APPEND(page);
+    ins = __col_insert_search(ins_head, cbt->ins_stack, cbt->next_stack, recno);
+    if (ins == NULL)
         cbt->compare = -1;
```

Comment:
You also need to set WT_CBT_VAR_ONPAGE_MATCH here (if there's no append list) because otherwise the code in cursor_valid will assume there's no on-disk value underneath the ins_head. However, because this code is reached for FLCS too, it needs to be conditional on the type.

(And at that point I'd move the assignments of cbt->ins_head and cbt->ins here too instead of above, seems tidier and avoids setting them twice in the common case.)

Reply:
Good call, thanks for spotting that. Agreed on moving the assignments around, I think the changes in af0ed4f should sort this out. At @quchenhao 's suggestion I've also added another test for when the entry on the append list is invisible in febb729.

Reply:
OK, the "request re-review" button is also broken (thanks github!) so here's an old-school notification: @sauclovian-wt this is ready for a re-review :-)

Reply:
Looks good. (Another consequence of whatever github stuff: I can't mark these resolved, so you'll need to unfortunately...)